http-server-netty/src/main/java/io/micronaut/http/server/netty/body/NettyBodyAdapter.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.server.netty.body;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.http.body.AvailableByteBody;
+import io.micronaut.http.body.ByteBody;
+import io.micronaut.http.netty.EventLoopFlow;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.EventLoop;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import reactor.core.publisher.Flux;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.LongUnaryOperator;
+
+/**
+ * Adapter from generic streaming {@link ByteBody} to {@link StreamingNettyByteBody}.
+ *
+ * @author Jonas Konrad
+ * @since 4.6.0
+ */
+@Internal
+public final class NettyBodyAdapter implements BufferConsumer.Upstream, Subscriber<ByteBuf> {
```

Comment:
Can some of those classes be package private?

Reply:
@yawkat 👆🏻